.circleci/scripts/cache.sh

```diff
@@ -0,0 +1,59 @@
+#!/bin/bash
+set -euo pipefail
+
+# Cache is available only on our custom runner
+if [ "$CIRCLECI_RUNNER_CLASS" != "spaceshard/ax41" ]; then
+    echo "Custom cache available only on self-hosted spaceshard/ax41 runner"
+    exit 0
+fi
+
+if [ $# -ne 1 ]; then
+    echo "Please provide cache action as an argument. (save, load or cleanup)"
+    exit 1
+fi
+
+action="$1"
+
+# VARIABLES
+# Files to use as cache key
+cache_key_files=("Cargo.lock" "rust-toolchain.toml") 
+
+ # Directories to cache
+cached_dirs=("target/release/.fingerprint target/release/build target/release/deps")
+
+# Cache files that are accessed more than $cache_cleanup_interval days ago will be removed in cleanup step
+cache_cleanup_interval=7 
+
+
+cache_base_dir="/cache" # dependent on runner architecture
+cache_file="$cache_base_dir/$(cat ${cache_key_files[@]} | shasum | awk '{print $1}').tar.gz"
+
+case "$action" in
+    "load")
+        echo "Loading cache..."
+        if [ ! -f "$cache_file" ]; then
+            echo "Cache does not exist."
+            exit 0
+        fi
+        tar xvf "$cache_file"
+        ;;
+    "save")
+        echo "Creating cache..."
+        if [ -f "$cache_file" ]; then
+            echo "Cache already exists."
+            exit 0
+        fi
+        tar czvf "$cache_file".tmp ${cached_dirs[@]} # Create a temporary cache file for atomicity
```

Comment:
Can you explain the atomicity?

Reply:
Maybe atomicity is not the most accurate description, this is to handle race conditions.`*tar.gz` file is created at the start of `tar x` command and it's being changed while `tar` is running. Since we have multiple runners which share this same cache, without this step, there is a possibility that one runner will try to read not completely written cache.

Reply:
Right, I didn't think about concurrent runs at all. You can add this explantation to the script for future readers. Is there a possibility that two runners will try writing at the same time, even into the tmp file? Is that a problem?

Reply:
Hm, that is a good point. Not sure, but better safe than sorry. I don't know how I missed half of the scenario. Fixing.

Reply:
Resolving converstaion after commit https://github.com/0xSpaceShard/starknet-devnet-rs/pull/559/commits/580cbb636b8fa5ef19b659a5ab7f3a4287413cec